### DISCUSSION TITLE
Getting airflow exception: DAG is missing start_date parameter

### DISCUSSION THREAD

**Discussion by AD97** [apache/airflow#29560] [Q&A] [answered]
When I checked the documentation, it was not mentioned anywhere that the start_date is a required parameter to be passed while creating a dag. I am getting the following error when I don't pass the parameter: 

```
Broken DAG: [/opt/airflow/dags/dag_api_to_minio.py] Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/baseoperator.py", line 1039, in dag
    dag.add_task(self)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/models/dag.py", line 2328, in add_task
    raise AirflowException("DAG is missing the start_date parameter")
airflow.exceptions.AirflowException: DAG is missing the start_date parameter
```

Here is my dag definition using the TaskFlow API:

```
@dag(start_date=None,
    schedule_interval=None, 
    max_active_runs=1,
    catchup=False,
    default_args={
        "retries": 1,
        "retry_delay": timedelta(minutes=3)
    },
    
) 
```

Versions: airflow:2.5.1, python:3.7

Can someone please explain what is going in here?

**Comment by josh-fell** [comment, marked as answer]
Hi @AD97! The `start_date` for a DAG ultimately needs to be a tangible value (e.g. `pendulum.datetime`, `airflow.utils.days_ago(3)`, etc.); however this value can be passed explicitly to the DAG constructor as `DAG(start_date=...)` or in the `default_args` for a DAG (i.e. `DAG(..., default_args={"start_date": ...)`). 

Perhaps the source of confusion is that the parameter is typed as `datetime | None` since the value can be passed in two different ways.

**Reply by AD97** [reply]
Thanks for that answer @josh-fell! If I understand it correctly, it is a required parameter that can be passed in two different ways. Does that mean we can suggest an edit in the official documentation saying the same? 
Here is the link for the section for which I'd like to suggest an edit: [https://docs.astronomer.io/learn/dags](https://docs.astronomer.io/learn/dags )

**Reply by Taragolis** [reply]
I guess preferable way is a explicit set on DAG level, because `default_args` for propagate arguments to task, just a couple of them use as fallback for DAG parameters.

BTW, this one is official documentation: https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#declaring-a-dag

**Reply by AD97** [reply]
Got it! Thanks, @Taragolis!
